### PR TITLE
UI report

### PR DESCRIPTION
**Pull request by ColeHiggins2**
Continuation of https://github.com/SatelliteQE/robottelo/pull/7637

### CODE REVIEW

File: tests/foreman/ui/test_reporttemplates.py [hunk 1]
```diff
@@ -342,3 +398,71 @@ def test_negative_nonauthor_of_report_cant_download_it(session):
...
+            result_json = session.reporttemplate.generate(
+                "Entitlements", values={'output_format': 'JSON'}
+            )
+            with open(result_json) as json_file:
+                data_json = json.load(json_file)
+            assert any(entitlement['Name'] == vm.hostname for entitlement in data_json)
+            assert any(
+                entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+                for entitlement in data_json
+            )
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_reporttemplates.py]: repeated three times, can't it go in some common function here? e.g. 

```python
def assert_reports_file( file_data, entitlement):
        assert any(entitlement['Name'] == vm.hostname for entitlement in file_data)
        assert any(
                entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
                for entitlement in file_data
            )
```

File: tests/foreman/ui/test_reporttemplates.py [hunk 2]
```diff
@@ -342,3 +398,71 @@ def test_negative_nonauthor_of_report_cant_download_it(session):
...
+                "Entitlements", values={'output_format': 'CSV'}
+            )
+            with open(result_csv) as csv_file:
+                data_csv = csv.DictReader(csv_file)
+                res = next(data_csv)
+            assert res['Name'] == vm.hostname
+            assert res['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+            result_html = session.reporttemplate.generate(
+                "Entitlements", values={'output_format': 'HTML'}
+            )
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_reporttemplates.py]: same as mentioned https://github.com/SatelliteQE/robottelo/pull/7651/files#r382418808

File: tests/foreman/ui/test_reporttemplates.py [hunk 3]
```diff
@@ -342,3 +398,71 @@ def test_negative_nonauthor_of_report_cant_download_it(session):
...
+            result_yaml = session.reporttemplate.generate(
+                "Entitlements", values={'output_format': 'YAML'}
+            )
+            with open(result_yaml) as yaml_file:
+                data_yaml = yaml.load(yaml_file, Loader=yaml.FullLoader)
+            assert any(entitlement['Name'] == vm.hostname for entitlement in data_yaml)
+            assert any(
+                entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+                for entitlement in data_yaml
+            )
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_reporttemplates.py]: same as mentioned https://github.com/SatelliteQE/robottelo/pull/7651/files#r382418808

File: tests/foreman/ui/test_reporttemplates.py [hunk 4]
```diff
@@ -342,3 +398,71 @@ def test_negative_nonauthor_of_report_cant_download_it(session):
...
     :CaseImportance: High
     """
+
+
+@tier3
+def test_positive_gen_entitlements_reports_multiple_formats(session, content_setup, module_org):
+    """Generate reports using the Entitlements template in html, yaml, json, and csv format.
+
+        :id: b268663d-c213-4e59-8f81-61bec0838b1e
+
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_reporttemplates.py]: extra space

**Comment by omkarkhatavkar** [comment]
@ColeHiggins2 Check the Travis failures

File: tests/foreman/ui/test_reporttemplates.py [hunk 5]
```diff
@@ -342,3 +398,71 @@ def test_negative_nonauthor_of_report_cant_download_it(session):
...
+                for entitlement in data_yaml
+            )
+            result_csv = session.reporttemplate.generate(
+                "Entitlements", values={'output_format': 'CSV'}
+            )
+            with open(result_csv) as csv_file:
+                data_csv = csv.DictReader(csv_file)
+                res = next(data_csv)
+            assert res['Name'] == vm.hostname
+            assert res['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
```
Comment (mirekdlugosz) [inline tests/foreman/ui/test_reporttemplates.py]: Haven't noticed that before. Technically, it checks only first row of CSV. To be consistent with rest of assertions and check for these strings anywhere in file, you would need:
```python
    res = list(data_csv)
assert any(entitlement['Name'] == vm.hostname for entitlement in res)
assert any(
    entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
    for entitlement in res
)
```
(maybe with better name than `res`)

File: tests/foreman/ui/test_reporttemplates.py (hunk 1 above)
Comment (JacobCallahan) [reply tests/foreman/ui/test_reporttemplates.py]: I think it's fine to be more explicit within the tests, so abstracting these checks away isn't much of a priority.

Comment (omkarkhatavkar) [reply tests/foreman/ui/test_reporttemplates.py]: I thought the assertion was repeating almost 4 times, with the same line. If I would be writing the code I would not repeat that code placed in a single place. I felt while I was reviewing the code. I won't take more than 10 mins.

**Comment by ColeHiggins2** [comment]
============================= test session starts ==============================
platform linux -- Python 3.6.6, pytest-4.6.3, py-1.8.1, pluggy-0.13.1 -- /home/[*******]/shiningpanda/jobs/ad0ef6b5/virtualenvs/d41d8cd9/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/[*******]/workspace/satellite6-standalone-automation
plugins: xdist-1.25.0, services-1.3.1, mock-1.10.4, forked-1.1.3, env-0.6.2
collecting ... 2020-02-24 14:43:06 - conftest - DEBUG - Collected 12 test cases
2020-02-24 09:43:07 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7ff350cc64a8
2020-02-24 09:43:07 - robottelo.ssh - INFO - Connected to [cisco-c240m3-01.khw2.lab.eng.bos.redhat.com]
2020-02-24 09:43:07 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-02-24 09:43:08 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-5.beta.el7sat.noarch

2020-02-24 09:43:08 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7ff350cc64a8
2020-02-24 09:43:08 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-02-24 09:43:08 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 12 items / 11 deselected / 1 selected

tests/foreman/ui/test_reporttemplates.py::test_positive_gen_entitlements_reports_multiple_formats PASSED [100%]

- generated xml file: /home/[*******]/workspace/satellite6-standalone-automation/foreman-results.xml -
================== 1 passed, 11 deselected in 278.59 seconds ===================

**Comment by JacobCallahan** [comment]
This has received enough ACKs to merge. Not adding mine, as I reviewed during development.

**Pull request merged**